src/command.rs

```diff
@@ -66,6 +74,8 @@ impl Command {
             }
           }
         }
+
+        rl.save_history(&history)?;
```

Comment:
Should we save the history after every command, so that in case of an error, the history still gets saved?

Reply:
Wouldn't `save_history` get called even if there was an error?

Reply:
If `command.apply` fails, for example, we would return from this function and `save_history` wouldn't get called.